.github/workflows/ci.yml

```diff
@@ -13,6 +13,130 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  build:
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        # the arm64 build takes too long, so disable for now
+        arch: [amd64]
+    name: Build Docker Image (${{ matrix.arch }})
+    runs-on: ubuntu-latest
+    env:
+      IMG_DEPS: ghcr.io/getsentry/relay-deps:${{ matrix.arch }}
+      # GITHUB_SHA in pull requests points to the merge commit
+      IMG_VERSIONED: ghcr.io/getsentry/relay:${{ matrix.arch }}-${{ github.event.pull_request.head.sha || github.sha }}
+      IMG_DEFAULT: ghcr.io/getsentry/relay:${{ github.event.pull_request.head.sha || github.sha }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - uses: swatinem/rust-cache@v2
+        with:
+          key: ${{ github.job }}-${{ matrix.arch }}
+
+      - run: docker run --rm --privileged tonistiigi/binfmt --install arm64
+        if: matrix.arch == 'arm64'
+
+      - name: ghcr.io login
+        run: docker login --username '${{ github.actor }}' --password '${{ secrets.GITHUB_TOKEN }}' ghcr.io
+
+      - name: build-deps
```

Comment:
Idea: These steps could be moved to a script that we can also use locally to:

- Build / update the builder image
- Build the binary to a local folder (should we introduce a `build/` directory?)
- Create the release docker container w/ the binary inside

In a close follow-up, we can then replace `docker-build-linux.sh` in the `build_binary` workflow with our new script.